### PR TITLE
Use `npe2 list` to show plugin info

### LINKED ISSUE #4456: napari --plugin-info does not list contributions from npe2 plugins
## 🐛 Bug

As noted by @tlambert03 in #4445, `napari --plugin-info` only lists hookimpl information about old-style napari_plugin_engine plugins, not npe2 plugins.

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
closes #4456 by using the new `npe2 list` to show plugin information with `napari --plugin-info` (as well as the plugins in `napari --info`

this needs a yet-to-be-released version of npe2, so is in draft for now

### CODE REVIEW

File: napari/__main__.py [hunk 1]
```diff
@@ -16,48 +16,28 @@
...
         print(sys_info())
-        from .plugins import plugin_manager
-
-        plugin_manager.discover_widgets()
-        if errors := plugin_manager.get_errors():
-            names = {e.plugin_name for e in errors}
-            print("\n!!  Errors were detected in the following plugins:")
-            print("(Run 'napari --plugin-info -v' for more details)")
-            print("\n".join(f"  - {n}" for n in names))
+        print("Plugins:")
```
Comment (nclack) [inline napari/__main__.py:25]: looks like the test_sys_info test is failing. Probably bc it's not grabbing the npe2 version with the list command?

**Review by nclack — COMMENTED** [review]
This looks good to me sans the test failure

File: napari/__main__.py (hunk 1 above)
Comment (tlambert03) [reply napari/__main__.py:25]: Well I actually removed plugins from this function. It’s still in the output “napari —info”.  But this function always seemed to be more about system specs.  Thoughts?

Comment (nclack) [reply napari/__main__.py:25]: Oh I see, `test_sys_info` is only looking at the string returned by `sys_info()` - Not what's being printed here.

I think that's fine.  The test should just get updated.

**Review by nclack — APPROVED** [review]

**Pull request merged**